packages/truffle-compile/parser.js

```diff
@@ -44,12 +43,7 @@ module.exports = {
       }
     };
 
-    var output = solc.compile(JSON.stringify(solcStandardInput), function() {
-      // The existence of this function ensures we get a parsable error message.
-      // Without this, we'll get an error message we *can* detect, but the key will make it easier.
-      // Note: This is not a normal callback. See docs here: https://github.com/ethereum/solc-js#from-version-021
-      return { error: importErrorKey };
-    });
```

Comment:
Oh I see. So if I get this right, the crux of the problem seems to be that our profiling relied on this weird error object, which was only available via solc-js?

Reply:
To be honest, I don't totally understand how this was even working. First of all, the object is named "errors" and not "error". Secondly, an almost same object is being returned by native solc as well but it did not work for the native solc object (the changes seemed to be in values and not keys so they should not matter).